java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -801,9 +802,21 @@ public CompletableFuture<GlideString[]> mget(@NonNull GlideString[] keys) {
     }
 
     @Override
-    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
-        String[] args = convertMapToKeyValueStringArray(keyValueMap);
-        return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+    public <ArgType> CompletableFuture<String> mset(@NonNull Map<ArgType, ArgType> keyValueMap) {
+        if (!keyValueMap.isEmpty()) {
+            var key = keyValueMap.keySet().iterator().next();
+            if (key instanceof String) {
+                String[] args = convertMapToKeyValueStringArray((Map<String, String>) keyValueMap);
+                return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+            }
+            if (key instanceof GlideString) {
+                GlideString[] args = flattenMapToGlideStringArray(keyValueMap);
+                return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+            } else {
+                throw new IllegalArgumentException("Expected String or GlideString");
+            }
+        }
+        return commandManager.submitNewCommand(MSet, new GlideString[0], this::handleStringResponse);
```

Comment:
add case in the integration test for empty map.
I don't think it should work...